### PR TITLE
[Intel GPU] DeepSeek V4 10/N : Add sqrtsoftplus support to fused_topk_torch_native

### PR DESCRIPTION
**Pull request by polisettyvarma**
## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #28629671419](https://github.com/sgl-project/sglang/actions/runs/28629671419)
Latest PR Test (Extra): :x: [Run #28629671320](https://github.com/sgl-project/sglang/actions/runs/28629671320)

### CODE REVIEW

File: python/sglang/srt/layers/moe/topk.py [hunk 1]
```diff
@@ -616,6 +616,8 @@ def scoring_func_impl(gating_output: torch.Tensor) -> torch.Tensor:
             return gating_output.softmax(dim=-1)
         elif scoring_func == "sigmoid":
             return gating_output.sigmoid()
+        elif scoring_func == "sqrtsoftplus":
```
Comment (jianan-gu) [inline python/sglang/srt/layers/moe/topk.py:653]: This is native path, and the changes is common for all device which use this func, suggest to correct the PR title (not XPU specific).  BTW, do we have UT to cover this change?

Comment (polisettyvarma) [reply python/sglang/srt/layers/moe/topk.py:653]: title is updated

Comment (polisettyvarma) [reply python/sglang/srt/layers/moe/topk.py:653]: UT added

File: test/manual/topk.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import torch
```
Comment (jianan-gu) [inline test/manual/topk.py]: can we check if there is existing UT for topk, and append testcase there? instead of adding a new file.

Comment (polisettyvarma) [reply test/manual/topk.py]: existing UT is on CPU - test/registered/cpu/test_topk.py

Comment (jianan-gu) [reply test/manual/topk.py]: ok, then the concern here is testing both native impl is weird : `native_biased_topk` vs `native_fused_topk`, would rather not adding such a UT.

Comment (polisettyvarma) [reply test/manual/topk.py]: removed UT

**Review by jianan-gu — APPROVED** [review]

**Review by mingfeima — APPROVED** [review]
